torch/distributed/distributed_c10d.py

```diff
@@ -3826,26 +3863,22 @@ def gather(tensor, gather_list=None, dst=0, group=None, async_op=False):
     else:
         gather_list = []
     _ensure_all_tensors_same_dtype(tensor, gather_list)
-
+    group = _group_or_default_group(group)
     if _rank_not_in_group(group):
         _warn_not_in_group("gather")
         return
-
-    my_rank = get_rank()
-    _validate_output_list_for_rank(my_rank, dst, gather_list)
-    output_tensors = [gather_list] if dst == my_rank else []
+    if dst is None and group_dst is None:
+        dst = 0
```

Comment:
My recommendation is to error out here, instead of making assumption, in case it becomes a support burden in future.

Reply:
oh, nvm, it is already a support burden..